pkg/server/server.go

```diff
@@ -26,9 +26,22 @@ import (
 	"github.com/spiffe/spire/pkg/server/svid"
 	"google.golang.org/grpc"
 
+	"errors"
+
+	"github.com/spiffe/spire/proto/server/datastore"
 	_ "golang.org/x/net/trace"
 )
 
+const (
+	invalidTrustDomainAttestedNode = "An attested node with trust domain '%v' has been detected, " +
+		"which does not match the configured trust domain of '%v'. If you want to change the trust domain, " +
+		"please delete all existing attested nodes"
```

Comment:
I don't think this is currently possible? https://github.com/spiffe/spire/issues/27 is related